tests/test_suite_validation.py

```
@@ -0,0 +1,699 @@
+    
+    def test_consistent_indentation(self, test_files):
+        """
+        Ensure every non-empty, non-comment line in the provided test files uses indentation in multiples of four spaces.
+        
+        Ignores blank lines and lines starting with `#`. On failure raises an assertion that includes the file name and line number.
+        
+        Parameters:
+            test_files (List[pathlib.Path]): Iterable of file paths to validate.
+        """
+        for test_file in test_files:
+            with open(test_file, 'r') as f:
```

Ruff (0.14.5)

Unnecessary mode argument

Remove mode argument

(UP015)